app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb

```diff
@@ -7,19 +7,47 @@ def teamdynamix_title
       SETTINGS[:teamdynamix][:title] || 'Team Dynamix'
     end
 
-    def teamdynamix_fields(host)
+    def teamdynamix_fields
       td_pane_fields = SETTINGS[:teamdynamix][:fields] || DEFAULT_TD_PANE_FIELDS
-      return [[_('Asset'), 'None Associated']] unless host.teamdynamix_asset_id
-      @asset = host.td_api.get_asset(host.teamdynamix_asset_id)
+      return [[_('Asset'), 'None Associated']] unless @host.teamdynamix_asset_id
+
+      get_teamdynamix_asset(@host.teamdynamix_asset_id)
+
+      # always display a link to the asset
+      fields = [ get_asset_uri ]
 
-      fields = []
       td_pane_fields.each do |field_name, asset_attr|
-        asset_attr_val = @asset[asset_attr]
-        fields += [[_(field_name.to_s), asset_attr_val]] if asset_attr_val.present?
+        asset_attr_val = @asset.has_key?(asset_attr) ? @asset[asset_attr] : get_nested_attrib_val(asset_attr)
+        fields += [[_(field_name.to_s), asset_attr_val]]
       end
       fields
     rescue StandardError => e
       [[_('Error'), e.message]]
     end
+
+    private
+    def get_asset_uri
+      api_url = SETTINGS[:teamdynamix][:api][:url]
+      asset_uri = api_url.split('api').first + @asset['Uri']
+      actual_link = "<a href='#{asset_uri}' target='_blank'>#{@asset['Uri']}</a>"
+      ['URI', actual_link.html_safe]
+    end
+
+    def get_teamdynamix_asset(asset_id)
+      @asset = @host.td_api.get_asset(asset_id)
+    rescue StandardError => e
+      raise "Error getting asset Data from Team Dynamix: #{e.message}"
+    end
+
+    def get_nested_attrib_val nested_attrib
+      parent_attrib, child_attrib = nested_attrib.split(".'")
```

Comment:
You don't need to use single string quotes in yaml. It will work fine if you just remove them.
Yaml parses strings with spaces just fine.

Reply:
I think for nested attributes like Attributes.'Ticket Routing Details' single quotes it makes it more readable and nicer. I am aware that yaml can parse strings and spaces. 

If you still want me to remove single quotes for nested attributes I can do that.

Reply:
Can you make your code handle both? So the end user can put string quotes or not in the yaml.